File: 977 Squares of a Sorted Array.cpp
class Solution {
public:
    // 双指针法，从头尾分别开始
    vector<int> sortedSquares(vector<int>& nums) {
        int i = 0;
        int j = nums.size() - 1;
        vector<int> new_nums(nums.size(), 0);
        for (int cnt = nums.size() - 1; cnt >= 0; cnt--) {
            if (abs(nums[j]) > abs(nums[i])) {
                new_nums[cnt] = pow(nums[j--], 2);
            } else {
                new_nums[cnt] = pow(nums[i++], 2);
            }   
        }
        return new_nums;
    }
};

File: 977 Squares of a Sorted Array.py
def sortedSquares(nums):
    left = 0
    right = len(nums) - 1
    new_num = [0] * len(nums)
    i = len(nums) - 1
    while(left < right):
        print(f"left: {left}, right: {right}, i: {i}")
        if(abs(nums[right]) > abs(nums[left])):
            new_num[i] = nums[right] ** 2
            i -= 1
            right -= 1
        elif(abs(nums[right]) < abs(nums[left])):
            new_num[i] = nums[left] ** 2
            i -= 1
            left += 1
        else:
            new_num[i] = nums[left] ** 2
            i -= 1
            left += 1
    new_num[0] = nums[left] ** 2
    return new_num
    
nums = [-7,-3,2,3,11]
print(sortedSquares(nums))